test/trento/domain/host/host_test.exs

```diff
@@ -522,4 +531,26 @@ defmodule Trento.HostTest do
       )
     end
   end
+
+  describe "event upcasting" do
+    test "should upcast HostRegistered event properly" do
+      assert %HostRegistered{
+               version: 2,
+               installation_source: :unknown
+             } ==
+               %{}
+               |> HostRegistered.upcast(%{})
+               |> HostRegistered.new!()
+    end
+
+    test "should upcast HostDetailsUpdated event properly" do
+      assert %HostDetailsUpdated{
+               version: 2,
+               installation_source: :unknown
+             } ==
+               %{}
+               |> HostDetailsUpdated.upcast(%{})
+               |> HostDetailsUpdated.new!()
+    end
```

Comment:
I would split these tests into two different files with the event module name to reflect the module structure (e.g. domain/hosts/events/host_details_updated.exs`

Also, the test is a bit unclear, we need to upcast a map that is at version 1, not an empty map because this is what we get from the event store when dealing with old events.

Something like 
```elixir
%{
                 "version" => 1,
# insert all the host details updated events fields that are present in version 1 possibly with a faker
....
               }
```

Reply:
or event without the version field, since it should default it at version 1